README.md

```diff
@@ -134,6 +137,69 @@ getWithShape({
 }).then( ... )
 ```
 
+### Resolving functions
+
+When the value is a function, its result will be used to resolve the shape.
+`falcor-shapes` accepts an optional second parameter; a resolution function called with `(<resolvable>, <shape>, <key>)`, where;
+
+* `resolvable` is the function specified in the shape
+* `shape` is the shape it is part of
+* `key` the function's key in the shape
+
+```js
+var shapeToSet = require('falcor-shapes')
+
+var sets = shapeToSet(
+  {
+    name: function () {
+      return {
+        first: true,
+        last: true
+      }
+    }
+  },
+  function(resolvable, shape, key) {
+    return resolvable()
+  }
+)
+
+model.get.apply(model, sets).then( ... )
+```
+
+A more complex example:
+
+```js
+var shapeToSet = require('falcor-shapes')
+
+var includeFirstName = true
+
+var sets = shapeToSet(
+  {
+    name: function (withFirstName) {
+      if (withFirstName) {
+        return {
+          first: true,
+          last: true
+        }
+      } else {
+        return {
+          last: true
+        }
+      }
+    }
+  },
+  function(resolvable, shape, key) {
+    if (key === 'name') {
+      return resolvable(includeFirstName)
```

Comment:
I'm a little confused on what this is adding, why not just reference `includeFirstName` in the method above?